lib/hocon/impl/simple_config_object.rb

```diff
@@ -107,6 +256,153 @@ def merged_with_object(abstract_fallback)
     end
   end
 
+  def modify(modifier)
+    begin
+      modify_may_throw(modifier)
+    rescue Hocon::ConfigError => e
```

Comment:
Good call.  I hadn't noticed that one.  Will push a change.  I'm thinking I may actually just get rid of those intermediate `RuntimeError` blocks, I don't think they can ever reasonably be hit.